src/test/java/co/cask/gcp/datastore/sink/DatastoreSinkConfigTest.java

```diff
@@ -0,0 +1,473 @@
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.gcp.datastore.sink.util.DatastoreSinkConstants;
+import co.cask.gcp.datastore.sink.util.IndexStrategy;
+import co.cask.gcp.datastore.sink.util.SinkKeyType;
+import co.cask.gcp.datastore.util.DatastorePropertyUtil;
+import com.google.cloud.datastore.PathElement;
+import com.google.datastore.v1.client.DatastoreHelper;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.spy;
+
+/**
+ * Tests of {@link DatastoreSinkConfig} methods.
+ */
+public class DatastoreSinkConfigTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testGetNamespaceNull() {
+    DatastoreSinkConfig config = DatastoreSinkConfigHelper.newConfigBuilder()
+      .setNamespace(null)
+      .build();
+
+    assertEquals(DatastorePropertyUtil.DEFAULT_NAMESPACE, config.getNamespace());
+  }
+
+  @Test
+  public void testGetNamespaceEmpty() {
+    DatastoreSinkConfig config = DatastoreSinkConfigHelper.newConfigBuilder()
+      .setNamespace("")
+      .build();
+
+    assertEquals(DatastorePropertyUtil.DEFAULT_NAMESPACE, config.getNamespace());
+  }
+
+  @Test
+  public void testGetNamespaceNotEmpty() {
+    String namespace = DatastoreSinkConfigHelper.TEST_NAMESPACE;
+    DatastoreSinkConfig config = DatastoreSinkConfigHelper.newConfigBuilder()
+      .setNamespace(namespace)
+      .build();
+
+    assertEquals(namespace, config.getNamespace());
+  }
+
+  @Test
+  public void testGetIndexedPropertiesNull() {
+    DatastoreSinkConfig config = DatastoreSinkConfigHelper.newConfigBuilder()
+      .setIndexedProperties(null)
+      .build();
+
+    assertEquals(Collections.emptySet(), config.getIndexedProperties());
+  }
+
+  @Test
+  public void testGetIndexedPropertiesEmpty() {
+    DatastoreSinkConfig config = DatastoreSinkConfigHelper.newConfigBuilder()
+      .setIndexedProperties("")
+      .build();
+
+    assertEquals(Collections.emptySet(), config.getIndexedProperties());
+  }
+
+  @Test
+  public void testGetIndexedPropertiesNotEmpty() {
+    String indexedProperties = "prop1,prop2,prop3";
+    DatastoreSinkConfig config = DatastoreSinkConfigHelper.newConfigBuilder()
+      .setIndexedProperties(indexedProperties)
+      .build();
+
+    Set<String> expectedProps = new TreeSet<>();
+    expectedProps.add("prop1");
+    expectedProps.add("prop2");
+    expectedProps.add("prop3");
+
+    Set<String> actualProps = config.getIndexedProperties();
+    assertNotNull(actualProps);
+
+    assertEquals(expectedProps, new TreeSet<>(actualProps));
```

Comment:
don't need to wrap it with a TreeSet, Set equality shouldn't depend on the order